internal/stage_write_tree.go

```diff
@@ -21,6 +21,8 @@ import (
 func testWriteTree(harness *test_case_harness.TestCaseHarness) error {
 	logger := harness.Logger
 	executable := harness.Executable
+	// This stage Requires the git binary for verifying the git object
```

Comment:
Seems weird behaviour to not do this here too. Can't we move the git binary back for verifying and then place it back again?

Reply:
Or can we just use the path of the temporary git to execute that?